db/dao/attempt_detail_dao.go

```diff
@@ -0,0 +1,54 @@
+package dao
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/webhookx-io/webhookx/db/entities"
+	"github.com/webhookx-io/webhookx/pkg/types"
+	"github.com/webhookx-io/webhookx/utils"
+)
+
+type attemptDetailDao struct {
+	*DAO[entities.AttemptDetail]
+}
+
+func NewAttemptDetailDao(db *sqlx.DB, workspace bool) AttemptDetailDAO {
+	return &attemptDetailDao{
+		DAO: NewDAO[entities.AttemptDetail]("attempt_details", db, workspace),
+	}
+}
+
+func (dao *attemptDetailDao) Upsert(ctx context.Context, attemptDetail *entities.AttemptDetail) error {
+	columns := make([]string, 0)
+	values := make([]interface{}, 0)
+	travel(attemptDetail, func(f reflect.StructField, v reflect.Value) {
```

Comment:
Nitpick: AttemptDetailDao should have all the knowledge about the attempt_details table including columns, so it's OK to use hard-coded SQL to compose the upsert statement, this would be more easy to review.

Reply:
oh, i thought this could be a generic method for DAO interface in the future.  
AttemptDetailDao use hard-coded SQL is ok, no problem.